packages/lesmis/src/api/queries/useVitalsData.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+import { useQuery } from 'react-query';
+import { get } from '../api';
+import { useEntityData } from './useEntityData';
+import { useEntitiesData } from './useEntitiesData';
+
+const getDecendantsOfType = (entities, rootEntityCode, type) => {
+  const entity = entities.find(e => e.code === rootEntityCode);
+
+  if (!entity) {
+    return [];
+  }
+
+  if (entity.type === type) {
+    return [entity.code];
+  }
+
+  if (!entity.childCodes) {
+    return [];
+  }
+  return entity.childCodes.map(c => getDecendantsOfType(entities, c, type)).flat();
+};
+
+const useSchoolInformation = entityCode => {
+  return useQuery(
+    ['vitals', 'school', entityCode],
+    () =>
+      get(`reportData/${entityCode}/LESMIS_school_vitals`, {
+        params: { endDate: '2021-01-01' },
+      }),
+    {
+      staleTime: 1000 * 60 * 60 * 1,
+      refetchOnWindowFocus: false,
+      retry: 1,
+      enabled: true,
+    },
+  );
+};
+
+const useDecendantSchoolInformation = (entities, entityCode) => {
+  const decendants = getDecendantsOfType(entities, entityCode, 'school');
+  return useQuery(
+    ['vitals', 'multi_school', entityCode],
+    () =>
+      get(`reportData/${entityCode}/LESMIS_multi_school_vitals`, {
+        params: { endDate: '2021-01-01', organisationUnitCodes: decendants.join() },
+      }),
+    {
+      staleTime: 1000 * 60 * 60 * 1,
+      refetchOnWindowFocus: false,
+      retry: 1,
+      enabled: true,
+    },
+  );
+};
+
+const useDistrictInformation = entityCode => {
+  return useQuery(
+    ['vitals', 'sub_district', entityCode],
+    () =>
+      get(`reportData/${entityCode}/LESMIS_sub_district_vitals`, {
+        params: { endDate: '2021-01-01' },
+      }),
+    {
+      staleTime: 1000 * 60 * 60 * 1,
+      refetchOnWindowFocus: false,
+      retry: 1,
+      enabled: true,
+    },
+  );
+};
+
+const useDecendantDistrictInformation = (entities, entityCode) => {
+  const decendants = getDecendantsOfType(entities, entityCode, 'sub_district');
+  return useQuery(
+    ['vitals', 'province', entityCode],
+    () =>
+      get(`reportData/${entityCode}/LESMIS_sub_district_vitals`, {
+        params: { endDate: '2021-01-01', organisationUnitCodes: decendants.join() },
+      }),
+    {
+      staleTime: 1000 * 60 * 60 * 1,
+      refetchOnWindowFocus: false,
+      retry: 1,
+      enabled: true,
+    },
+  );
+};
+
+export const useVitalsData = entityCode => {
+  const { data: entityData } = useEntityData({ entityCode });
+  const { data: entities = [] } = useEntitiesData();
```

Comment:
```suggestion
  const { data: entities = [], ...entityQuery } = useEntitiesData();
  const entityData = entities.find(e => e.code === entityCode);
```
I would get the current entity from the entities data to potentially save a query and pull out the rest of the react query object to be returned later in `entityQuery`